plugins/fmcomms5.c

```diff
@@ -1697,7 +1697,8 @@ static void save_widgets_to_ini(FILE *f)
 			!!gtk_toggle_tool_button_get_active(section_toggle[SECTION_GLOBAL]),
 			!!gtk_toggle_tool_button_get_active(section_toggle[SECTION_TX]),
 			!!gtk_toggle_tool_button_get_active(section_toggle[SECTION_RX]),
-			!!gtk_toggle_tool_button_get_active(section_toggle[SECTION_FPGA]));
+			!!gtk_toggle_tool_button_get_active(section_toggle[SECTION_FPGA])) == sizeof(buf))
+		return;
 	fwrite(buf, 1, strlen(buf), f);
```

Comment:
Maybe we can change all of these that follow the pattern of
```
  snprintf(buf, ...);
  fwrite(buf, strlen(buf), f);
```
to
```
fprintf(f, ...);
```

That should be equivalent, but not require the tmp buffer.

Reply:
cocci script to help with this
```
@@
expression buf;
expression f;
@@
+fprinf(f,
-snprintf(buf, sizeof(buf),
 ...);
-fwrite(buf, ..., f);
```